contrib/confluentinc/confluent-kafka-go/kafka/kafka.go

```diff
@@ -354,8 +354,8 @@ func (p *Producer) Produce(msg *kafka.Message, deliveryChan chan kafka.Event) er
 
 	setProduceCheckpoint(p.cfg.dataStreamsEnabled, p.libraryVersion, msg)
 	err := p.Producer.Produce(msg, deliveryChan)
-	// with no delivery channel, finish immediately
-	if deliveryChan == nil {
+	// with no delivery channel or enqueue error, finish immediately
+	if err != nil || deliveryChan == nil {
```

Comment:
This makes sense to me, it seems like in case of an error we'd never close the span. Does this change also need to be performed in the kafka.v2 version? https://github.com/DataDog/dd-trace-go/blob/55ad19bd3abb08174613c2261856f310b3d442a3/contrib/confluentinc/confluent-kafka-go/kafka.v2/kafka.go#L356-L360

Reply:
Good catch!